src/secure_channel.rs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2022 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+//! Agent is a module used to connect KBS (Relying Party)
+//! for confidential resources
+
+use std::collections::HashMap;
+
+use anyhow::*;
+use serde::{Deserialize, Serialize};
+use tokio::fs;
+use tonic::transport::Channel;
+
+use self::get_resource::{
+    get_resource_service_client::GetResourceServiceClient, GetResourceRequest,
+};
+
+mod get_resource {
+    #![allow(unknown_lints)]
+    #![allow(clippy::derive_partial_eq_without_eq)]
+    tonic::include_proto!("getresource");
+}
+
+/// Attestation Agent's GetResource gRPC address.
+/// It's given <https://github.com/confidential-containers/attestation-agent#run>
+pub const AA_GETRESOURCE_ADDR: &str = "http://127.0.0.1:50001";
+
+/// The resource description that will be passed to AA when get resource.
+#[derive(Serialize, Deserialize, Debug)]
+struct ResourceDescription {
+    name: String,
+    optional: HashMap<String, String>,
+}
+
+impl ResourceDescription {
+    /// Create a new ResourceDescription with resource name.
+    pub fn new(name: &str, optional: HashMap<String, String>) -> Self {
+        ResourceDescription {
+            name: name.to_string(),
+            optional,
+        }
+    }
+}
+
+/// SecureChannel to connect with KBS
+pub struct SecureChannel {
+    /// Get Resource Service client.
+    client: GetResourceServiceClient<Channel>,
+    kbc_name: String,
+    kbc_uri: String,
```

Comment:
If I understand correctly, the `kbc_uri` is same as the `KBS_URI` in the [docs](https://github.com/confidential-containers/attestation-agent/blob/main/docs/IMPLEMENTATION.md) of the attestation-agent. Should we follow the convention, converting the  `kbc_uri`  to `kbs_uri`?

Reply:
Yes, thanks! Fixed it